src/monster.cpp

```diff
@@ -458,12 +449,54 @@ int monster::print_info(WINDOW* w, int vStart, int vLines, int column) const
 
 std::string monster::extended_description() const
 {
-    // @todo Add tons of info here
-    std::string ret = string_format( _ ( "This is a %s" ), name().c_str() );
-    ret += "\n--\n";
+    std::ostringstream ss;
+
+    nc_color color = c_white;
+    std::string attitude = "";
+    get_Attitude( color, attitude );
+    attitude = get_tag_from_color( color ) + attitude;
+
+    ss << string_format(
+           _( "This is a %s %s <stat>%s</stat>" ),
+           name().c_str(),
+           attitude.c_str(),
+           get_effect_status().c_str()
+       ) << std::endl;
+
+    ss << "--" << std::endl;
     auto hp_bar = hp_description( hp, type->hp );
-    ret += get_tag_from_color( hp_bar.second ) + hp_bar.first;
-    return replace_colors( ret );
+    ss << get_tag_from_color( hp_bar.second ) << hp_bar.first << std::endl;
+
+    ss << "--" << std:: endl;
+    ss << string_format( "<dark>%s</dark>", type->description.c_str() ) << std::endl;
+
+    ss << "--" << std::endl;
+
+    if( type->has_flag( m_flag::MF_HEARS ) ) {
+        ss << _( "It can hear" ) << std::endl;
+    } else {
+        ss << _( "It can't hear" ) << std::endl;
+    }
+
+    if( type->has_flag( m_flag::MF_SEES ) ) {
+        ss << _( "It can see" ) << std::endl;;
+    } else {
+        ss << _( "It can't see" ) << std::endl;
+    }
+
+    if( type->has_flag( m_flag::MF_SMELLS ) ) {
+        ss << _( "It can sense smell" ) << std::endl;
+    } else {
+        ss << _( "It can't sense smell" ) << std::endl;
+    }
+
+    if( type->has_flag( m_flag::MF_NOHEAD ) ) {
+        ss << _( "It doesn't have head" ) << std::endl;
+    } else {
+        ss << _( "It has head" ) << std::endl;
+    }
+
```

Comment:
For the flags, I'd only add one note - the positive one.

For senses, it may be worth merging them into one line: "It has the following senses: sight, and smell" (using `enumerate_as_string`)